### PR TITLE
Fix for batch size != 1

### PR DESCRIPTION
**Pull request by CoderHam**
Fix for #544. Batch size is now taken into account for Inputs in shared memory

### CODE REVIEW

File: src/clients/c++/request_grpc.cc [hunk 1]
```diff
@@ -854,7 +854,8 @@ InferGrpcContextImpl::PreRunProcessing(std::shared_ptr<Request>& request)
       rshared_memory->set_offset(
           reinterpret_cast<InputImpl*>(io.get())->GetSharedMemoryOffset());
       rshared_memory->set_byte_size(
-          reinterpret_cast<InputImpl*>(io.get())->ByteSize());
+          reinterpret_cast<InputImpl*>(io.get())->ByteSize() *
```
Comment (deadeyegoodwin) [inline src/clients/c++/request_grpc.cc]: This should be TotalByteSize... but you've changed that to mean just the total size of the raw bytes. I think you need to fix TotalByteSize to be what it originally was... the total size of an input (i.e. entire batch). See above the call to set_batch_byte_size() is going to be wrong for the shared memory case (and that is needed to correctly support string datatype. You also need a TotalSendByteSize (for http content size and maybe something else but you likely need to make another class member for that).

Comment (CoderHam) [reply src/clients/c++/request_grpc.cc]: Done

File: src/clients/c++/request_common.h [hunk 2]
```diff
@@ -157,6 +157,7 @@ class InputImpl : public InferContext::Input {
   size_t GetSharedMemoryOffset() const { return shm_offset_; }
 
   void SetBatchSize(size_t batch_size) { batch_size_ = batch_size; }
+  size_t TotalSendByteSize() const { return total_send_byte_size_; }
```
Comment (deadeyegoodwin) [inline src/clients/c++/request_common.h]: Move this up under TotalByteSize()

Comment (CoderHam) [reply src/clients/c++/request_common.h]: Done

**Pull request merged**